packages/p2p/src/pages/buy-sell/buy-sell-form.jsx

```diff
@@ -20,14 +20,16 @@ import PaymentMethodIcon from 'Components/payment-method-icon';
 import './buy-sell-form.scss';
 
 const BuySellForm = props => {
-    const isMounted = useIsMounted();
     const { advertiser_page_store, buy_sell_store, general_store, my_profile_store } = useStores();
     const [selected_methods, setSelectedMethods] = React.useState([]);
-    buy_sell_store.setFormProps(props);
     const { showModal } = useModalManagerContext();
     const { data: p2p_advertiser_payment_methods } = useP2PAdvertiserPaymentMethods();
 
-    const { setPageFooterParent } = props;
+    React.useEffect(() => {
+        buy_sell_store.setFormProps(props);
+    }, [props, buy_sell_store]);
+
+    const { advert, setIsSubmitDisabled, setPageFooterParent, setSubmitForm } = props;
```

Comment:
here is the slight difference, where i destructured the props and then call them below instead of directly calling them from `set_form_props` in `buy_sell_store`